### PR TITLE
fix(HTTP Request Tool Node): Respond with an error when receive binary response

### PR DESCRIPTION
**Pull request by burivuhster**
## Summary

This PR fixes the HTTP Tool issue with binary responses. Now tool will return a "Binary data is not supported" error back to the model, instead of the raw bytes.

## Related Linear tickets, Github issues, and Community forum posts

- https://linear.app/n8n/issue/AI-378/community-issue-tool-that-returns-binary-borks-ai-agent-add-option-to
- https://github.com/n8n-io/n8n/issues/11069

## Review / Merge checklist

- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/@n8n/nodes-langchain/nodes/tools/ToolHttpRequest/utils.ts [hunk 1]
```diff
@@ -749,11 +749,29 @@ export const configureToolFunction = (
...
 			}
+
+			if (!response) {
+				try {
+					// Check if the response is binary data
+					if (fullResponse?.headers?.['content-type']) {
+						const contentType = fullResponse.headers['content-type'] as string;
+						const mimeType = contentType.split(';')[0].trim();
+
+						if (mime.charset(mimeType) !== 'UTF-8') {
```
Comment (jeanpaul) [inline packages/@n8n/nodes-langchain/nodes/tools/ToolHttpRequest/utils.ts:765]: What happens if the charset is a foreign charset? Does `httpRequest` convert it?

Comment (burivuhster) [reply packages/@n8n/nodes-langchain/nodes/tools/ToolHttpRequest/utils.ts:765]: It takes the default charset for a given mime type, which should be 'UTF-8' for all text mime types

**Review by jeanpaul — APPROVED** [review]

File: packages/@n8n/nodes-langchain/nodes/tools/ToolHttpRequest/test/ToolHttpRequest.node.test.ts [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+			const res = await (response as N8nTool).invoke('');
+			expect(res).toEqual('Hello World');
+		});
+
+		it('should return the response text when receiving a text response with a charset', async () => {
+			mockRequest.mockResolvedValue({
+				body: 'こんにちは世界',
+				headers: {
+					'content-type': 'text/plain; charset=iso-2022-jp',
```
Comment (jeanpaul) [inline packages/@n8n/nodes-langchain/nodes/tools/ToolHttpRequest/test/ToolHttpRequest.node.test.ts:113]: Thanks for adding this test! 🙏

**Comment by janober** [comment]
Got released with [`n8n@1.64.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.64.0)

**Pull request merged**